packages/perseus-core/src/parse-perseus-json/regression-tests/parse-perseus-json-regression.test.ts

```diff
@@ -144,6 +145,47 @@ describe("parseAndMigratePerseusArticle", () => {
     });
 });
 
+describe("parseAndMigrateUserInputMap", () => {
+    beforeAll(() => {
+        registerCoreWidgets();
+    });
+
+    describe.each(userInputDataFiles)("given %s", (filename) => {
+        const json = fs.readFileSync(join(userInputDataDir, filename), "utf-8");
+        const result = parseAndMigrateUserInputMap(json);
+
+        it("parses successfully", () => {
+            // If the parse fails, get just the error message. This makes the test
+            // failure easier to read, since otherwise the entire `invalidObject`
+            // from the ParseFailureDetail would be printed.
+            const resultWithMessage = mapFailure(getMessage)(result);
+
+            expect(resultWithMessage).toEqual(anySuccess);
+        });
+
+        it("returns the same result as before", () => {
+            assertSuccess(result);
+            expect(result.value).toMatchSnapshot();
```

Comment:
We talked about this in person, but just for the paper trail: I personally think it would benefit us to move away from these kinds of raw JSON / snapshot tests. It's hard to immediately understand the test (inputs/outputs/expectations) and they're really easy to auto-update without understanding the implications of the changes.

I think Ben landed on keeping these as-is while considering a refactor away from these kinds of tests. I'm also fine to have these exist as-is, so I'll approve regardless.